### PR TITLE
`PMP::isotropic_remeshing()` -  fix relaxation of constrained vertices

### LINKED ISSUE #8388: params::vertex_is_constrained_map setting in PMP::isotropic_remeshing does not set the corner points remain its position during remesh
_Please use the following template to help us solving your issue._

## Issue Details

I am using PMP::isotropic_remeshing method to remesh a surface mesh. I have a vertex_is_constrained_map in which the corner points are set to be true.  

Before isotropic_remeshing, the surface mesh is as follows:
image

and the green points are corner points (for these vertices, vertex_is_constrained_map[vertex] = true )

However, after isotropic_remeshing, the corner points are not at the same position, which is not what I desire:
image
image

I use isotropic_remeshing to another example but that example works fine: the corner points all remains at its position:
image
I dont understand why the two isotropic_remeshing (I implement the two almost the same) results differently.

## Source Code

```
//only part of the code is shown, because the code is a little long.

    std::vector<face_descriptor> sp_selected_faces;
    std::vector<bool> sp_is_selected(num_faces(mesh2_clip), false);
    for (edge_descriptor e : edges(mesh2_clip))
        if (sp_constrained_map[e])
        {
            // insert all faces incident to the target vertex
            for (halfedge_descriptor h : halfedges_around_target(halfedge(e, mesh2_clip), mesh2_clip))
            {
                if (!is_border(h, mesh2_clip))
                {
                    face_descriptor f = face(h, mesh2_clip);
                    if (!sp_is_selected[f])
                    {
                        sp_selected_faces.push_back(f);
                        sp_is_selected[f] = true;
                    }
                }
            }
        }
// increase the face selection
CGAL::expand_face_selection(sp_selected_faces, mesh2_clip, 2,
    Vector_pmap_wrapper(sp_is_selected), std::back_inserter(sp_selected_faces));

std::cout << sp_selected_faces.size()
    << " faces were selected for the remeshing step\n";


Mesh::Property_map<vertex_descriptor, bool> vertex_is_constrained_split =
    mesh2_clip.add_property_map<vertex_descriptor, bool>("v:is_constrained", false).first;

for each (vertex_descriptor v in v_constrained_vec_split)
{
    vertex_is_constrained_split[v] = true;
}

PMP::isotropic_remeshing(sp_selected_faces, target_edge_length, mesh2_clip, params::vertex_is_constrained_map(vertex_is_constrained_split));

mesh2_clip.collect_garbage();

CGAL::draw(mesh2_clip);
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits):
* Compiler:
* Release or debug mode:
* Specific flags used (if any):
* CGAL version:
* Boost version:
* Other libraries versions if used (Eigen, TBB, etc.):

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

Constrained vertices, and constrained edges were not properly protected during the smoothing step.

## Release Management

* Affected package(s): PMP
* Issue(s) solved (if any): fix #8388
* License and copyright ownership: unchanged

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h [hunk 1]
```diff
@@ -1364,6 +1356,24 @@ namespace internal {
...
+      {
+        if (is_on_patch(h))
+          continue;
+        else if (is_on_patch_border(h) && relax_constraints)
+          continue;
+        else
+          return false;
+      }
+      return true;
+    }
```
Comment (lrineau) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h:1375]: `border_halfedges`, line 1364, does not seem to be used.

**Comment by sloriot** [comment]
I wonder if your changes could be responsible for the runtime error:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-19/Surface_mesh_approximation/TestReport_Christo_MSVC-2022-Preview-Release.gz

**Comment by janetournois** [comment]
Indeed the test fails in my branch, not in master.
Thank you for the report

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.1-Ic-22

**Comment by sloriot** [comment]
@janetournois still a draft or ready to be merged?

**Comment by janetournois** [comment]
Hello @citystrawman
this PR should solve your issue #8388 
Can you please give it a try so check that the problem is solved on your side?
Thanks!

**Comment by citystrawman** [comment]
> Hello @citystrawman this PR should solve your issue #8388 Can you please give it a try so check that the problem is solved on your side? Thanks!

Hi @janetournois I downloaded 5.6.x-branch but failed to build polygon mesh processing examples, here's the error info when configuring in CMake:
image
```
CMake Error at CMakeLists.txt:82 (include):

  include could not find requested file:

    UseOpenMesh


Can not locate METIS include directory
Can not locate METIS library
NOTICE: Examples that use the METIS library will not be compiled.
CMake Warning (dev) at D:/softwares/vcpkg/installed/x64-windows/share/cgal/FindTBB.cmake:275 (message):
  Unrecognized MSVC version (1942).  Please update FindTBB.cmake.  Some TBB_*
  CMake variables may need to be set manually.
Call Stack (most recent call first):
  D:/softwares/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  CMakeLists.txt:118 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

I have installed openmesh using vcpkg. Do you know how to solve it?

**Comment by citystrawman** [comment]
> Hello @citystrawman this PR should solve your issue #8388 Can you please give it a try so check that the problem is solved on your side? Thanks!

Hi @janetournois , I just downloaded the latest 5.6.x-branch but I failed to link the project with `CGAL::CGAL_Basic_viewer`. I have already downloaded qt5 and cgal[qt] using vcpkg, but when I configure the CMakeLists.txt I got errors:
image

And here's the CMakeLists.txt snippet:

```
message(STATUS "===========Finding Qt5===========")
#CGAL_Qt5 is needed for the drawing.
#set(Qt5_DIR "D:/softwares/vcpkg/installed/x64-windows/share/qt5")
find_package(CGAL REQUIRED OPTIONAL_COMPONENTS Qt5)

if(CGAL_Qt5_FOUND)
  message(STATUS "Link CGAL_Basic_viewer to projects.")
  target_link_libraries(clip_get_intersect_points_new_try_remove2 PUBLIC CGAL::CGAL_Basic_viewer)
else()
  message(STATUS "===========NOTICE: Qt5 not found===========")
endif()
```
the printed out info is :
```
2>-- ===========Finding Qt5===========
2>CMake Error at D:/softwares/vcpkg/installed/x64-windows/share/cgal/CGALConfig.cmake:138 (message):
2>  The requested CGAL component Qt5 does not exist!
2>Call Stack (most recent call first):
2>  D:/softwares/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
2>  CMakeLists.txt:85 (find_package)
2>
2>
2>-- Configuring incomplete, errors occurred!
2>D:\Program Files\Microsoft Visual Studio\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'E:\Work\202501\cgal-5.6.x-branch\Polygon_mesh_processing\examples\Polygon_mesh_processing\CMakeLists.txt' exited with code 1.
2>Done building project "clip_get_intersect_points_new_try_remove2.vcxproj" -- FAILED.
```

**Comment by janetournois** [comment]
There may be an inconsistency between your vcpkg CGAL version (probably 6.0 if you uploaded it recently, which works with Qt6), and the 5.6.x-branch, which works with Qt5.
You can use the master branch of CGAL (which also has the fix), or you can try to locally hack the CMakeLists you mention by replacing occurrences of "Qt5" by "Qt6".

**Comment by citystrawman** [comment]
> There may be an inconsistency between your vcpkg CGAL version (probably 6.0 if you uploaded it recently, which works with Qt6), and the 5.6.x-branch, which works with Qt5. You can use the master branch of CGAL (which also has the fix), or you can try to locally hack the CMakeLists you mention by replacing occurrences of "Qt5" by "Qt6".

somehow I failed install Qt6 by vcpkg, and I gave up using the browser. However, for the slope model, the constrained vertices after remeshing is still incorrect:
image
image
I tried both 5.6.x branch as well as the master branch.

**Pull request merged**